native_lib/test_user_setup.c

```diff
@@ -0,0 +1,136 @@
+                                        DDGSYNCCRYPTO_STRETCHED_PRIMARY_KEY_SIZE,
+                                        DDGSyncCryptoStretchedPrimaryKeySubkey,
+                                        DDGSYNC_STRETCHED_PRIMARY_KEY_CONTEXT,
+                                        primaryKey)) {
+
+        return DDGSYNCCRYPTO_CREATE_STRETCHED_PRIMARY_KEY_FAILED;
+    }
+
+    randombytes_buf(nonceBytes, crypto_secretbox_NONCEBYTES);
+
+    if (0 != crypto_secretbox_easy(protectedSecretKey,
+                                   secretKey,
+                                   DDGSYNCCRYPTO_SECRET_KEY_SIZE,
+                                   nonceBytes,
+                                   stretchedPrimaryKey)) {
+        return DDGSYNCCRYPTO_CREATE_PROTECTED_SECRET_KEY_FAILED;
+    }
+
+    memcpy(&protectedSecretKey[crypto_secretbox_MACBYTES + DDGSYNCCRYPTO_SECRET_KEY_SIZE], nonceBytes, crypto_secretbox_NONCEBYTES);
+
+    return DDGSYNCCRYPTO_OK;
+}
+
+int main(int argc, char **argv) {
+    if (argc != 4) {
+        printf("Usage: %s secretkey username password\n", argv[0]);
+        return 1;
+    }
+
+    unsigned char primaryKey[DDGSYNCCRYPTO_PRIMARY_KEY_SIZE];
+    unsigned char secretKey[DDGSYNCCRYPTO_SECRET_KEY_SIZE];
+    unsigned char protectedSecretKey[DDGSYNCCRYPTO_PROTECTED_SECRET_KEY_SIZE];
+    unsigned char passwordHash[DDGSYNCCRYPTO_HASH_SIZE];
+    
+    DDGSyncCryptoResult res;
+    {
+        int err = sodium_base642bin(secretKey, DDGSYNCCRYPTO_SECRET_KEY_SIZE, argv[1], strlen(argv[1]), NULL, NULL, NULL, sodium_base64_VARIANT_ORIGINAL);
+        if (err != 0) {
+            printf("err %d b64 decoding secretKey\n", err);
+            return 2;
+        }
+    }
+
+    res = ddgSyncGenerateAccountKeysWithSecretKey(primaryKey, secretKey, protectedSecretKey, passwordHash, argv[2], argv[3]);
+    if (res != 0) {
+        printf("errorcode: %d\n", res);
+        return 3;
+    }
+
+    printf("{\n");
+    {
+        unsigned char printbuf[sizeof(primaryKey)*2];
+        sodium_bin2base64((char*)printbuf, sizeof(printbuf), primaryKey, sizeof(primaryKey), sodium_base64_VARIANT_ORIGINAL);
+        printf("\t{ \"primaryKey\": \"%s\",\n", printbuf);
+    }
+    {
+        unsigned char printbuf[sizeof(secretKey)*2];
+        sodium_bin2base64((char*)printbuf, sizeof(printbuf), secretKey, sizeof(secretKey), sodium_base64_VARIANT_ORIGINAL);
+        printf("\t{ \"secretKey\": \"%s\"\n,", printbuf);
+    }
+    {
+        unsigned char printbuf[sizeof(protectedSecretKey)*2];
+        sodium_bin2base64((char*)printbuf, sizeof(printbuf), protectedSecretKey, sizeof(protectedSecretKey), sodium_base64_VARIANT_ORIGINAL);
+        printf("\t{ \"protectedSecretKey\": \"%s\"\n,", printbuf);
+    }
+    {
+        unsigned char printbuf[sizeof(passwordHash)*2];
+        sodium_bin2base64((char*)printbuf, sizeof(printbuf), passwordHash, sizeof(passwordHash), sodium_base64_VARIANT_ORIGINAL);
+        printf("\t{ \"passwordHash\": \"%s\"\n,", printbuf);
```

Comment:
if I'm reading this right, you're adding some extra `{` that shouldn't be there. Or am I mistaken?

```suggestion
        printf("  \"passwordHash\": \"%s\"\n,", printbuf);
```

Reply:
fixed.  The output passes through JQ now.